### ISSUE TITLE
Problem with GAMA and CityIO coupling

### ISSUE THREAD

**Issue by agrignard** [yasushisakai/CityIOServer#128]
We have in this repo https://github.com/CityScope/CS_Simulation_GAMA/tree/master/CS_CityScope_GAMA/models/cityIO some example of working bridge between cityIo and GAMa that was working pretty well and documented here https://cityscope.media.mit.edu/modules/GAMA.

It seems to have been broken in the recent days or weeks. 

The get is still working but the post seems not to work anymore.
Gama used to send information as a url to to cityIO before using this syntax for instance
save(json_file("https://cityio.media.mit.edu/api/table/update/"+city_io_table+"/"+GEOGRID));

@yasushisakai did something change in the way to post information?

**Comment by yasushisakai** [comment]
We've been preparing for a **breaking change** version since late March and Brix (python) and the frontend (javascript) have been migrated to this new version. The server is now running with this specification.

You can find the endpoint specs [here](https://cityscope.media.mit.edu/backend/API). 

As for the specific post endpoint:
On top of new functionallity, past server crashes came from weak specifications of the API(s), so generally the rules for the endpoints are slightly stricter. From the top of my mind, there are three things you need to change. 

1. Add a trailing slash at the end `/`
2. Make sure you have the `Content-Type:application/json` in your request header. If not, the server will interpret the payload as bytes not json.
3. Remove `update` term from endpoint

to sum this up, the following curl command will pass what you intend, again this is specified [here](https://cityscope.media.mit.edu/backend/API#post-a-table)

```
curl -X POST -d '{"data":"what ever"}' -H "Content-Type:application/json" https://cityio.media.mit.edu/api/table/test_table/
```

I will update the above docs that needs an catch up.

**Comment by yasushisakai** [comment]
@agrignard does this need follow-up?

**Comment by agrignard** [comment]
I am working on it but I still cannot figure exaclty how to do that from GAMA.

This what I have so far in GAMA
````
save(json_file("https://cityio.media.mit.edu/api/table/"+city_io_table+"/"+type, m)) header:true;
````
In Gama The save function can specify if there is a header or not as a boolean but I cannot see how to add the string ```Content-Type:application/json```shoudl I simply concatenate this string in the json that I sent?

What I don't understand is that when I open a table like this one for instance https://cityio.media.mit.edu/api/table/test/ I do'nt see the string ```Content-Type:application/json```

**Comment by agrignard** [comment]
So it seems to work more or less with this
```save(json_file("https://cityio.media.mit.edu/api/table/"+city_io_table+"/"+type, m)) header:true;```

But I am not specifying the ```Content-Type:application/json``` can it still be working? without this information?

**Comment by yasushisakai** [comment]
@agrignard The server will reject the request if you don't have that header. Since you see the table posted, I assume it's handled by the underlying library.

**Comment by yasushisakai** [comment]
closing! feel free to re-open.

**Comment by agrignard** [comment]
OK cool ! 
so let's assume it works then (evn thogh I am not specifying it by hand in Gama but as you said myabe it's handle by the library
